deepchecks/vision/checks/performance/single_dataset_scalar_performance.py

```diff
@@ -0,0 +1,121 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing a single dataset with a signle scalar output performance check."""
+import typing as t
+import torch
+from ignite.metrics import Metric, Accuracy
+from deepchecks.core import CheckResult, ConditionResult, DatasetKind
+from deepchecks.core.condition import ConditionCategory
+from deepchecks.vision import Batch, Context, SingleDatasetCheck
+from deepchecks.core.errors import DeepchecksValueError
+from deepchecks.vision.vision_data import TaskType
+from deepchecks.vision.metrics_utils.object_detection_precision_recall import \
+    ObjectDetectionAveragePrecision
+import warnings
+
+__all__ = ['SingleDatasetScalarPerformance']
+
+
+class SingleDatasetScalarPerformance(SingleDatasetCheck):
+    """Summarize a given metric over a given dataset to return a performance score as a scalar.
```

Comment:
```suggestion
    """Calculate a scalar performance metrics for a given model and a given dataset.
```
Just a suggestion for improvement, you can think perhaps of something else